source/common/http/match_delegate/config.cc

```diff
@@ -319,6 +319,58 @@ Envoy::Http::FilterFactoryCb MatchDelegateConfig::createFilterFactoryFromProtoTy
   };
 }
 
+absl::StatusOr<Envoy::Http::FilterFactoryCb> MatchDelegateConfig::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::common::matching::v3::ExtensionWithMatcher& proto_config,
+    const std::string& prefix, DualInfo, Server::Configuration::UpstreamFactoryContext& context) {
+
+  ASSERT(proto_config.has_extension_config());
+  auto& factory =
+      Config::Utility::getAndCheckFactory<Server::Configuration::UpstreamHttpFilterConfigFactory>(
+          proto_config.extension_config());
+
+  auto message = Config::Utility::translateAnyToFactoryConfig(
+      proto_config.extension_config().typed_config(),
+      context.serverFactoryContext().messageValidationVisitor(), factory);
+  auto filter_factory_or_error = factory.createFilterFactoryFromProto(*message, prefix, context);
+  if (!filter_factory_or_error.ok()) {
+    throwEnvoyExceptionOrPanic(std::string(filter_factory_or_error.status().message()));
+  }
+  auto filter_factory = filter_factory_or_error.value();
+
+  Factory::MatchTreeValidationVisitor validation_visitor(*factory.matchingRequirements());
+
+  Envoy::Http::Matching::HttpFilterActionContext action_context{prefix, absl::nullopt,
```

Comment:
Is this an issue to pass in an absl::nullopt into this data object?  @wbpcode

Reply:
The reason we pass in an absl::nullopt is due to there is no Server::Configuration::FactoryContext object can be retrieved here.